### DISCUSSION TITLE
resources bundled via spec file not found upon execution

### DISCUSSION THREAD

**Discussion by metaperl** [pyinstaller/pyinstaller#7095] [PyInstaller]
i am using [this spec file](https://github.com/metaperl/flash-goban/blob/main/myspec.spec) to build a windows executable. when running the executable, i get this error;

```
Traceback (most recent call last):
  File "main.py", line 190, in <module>
  File "traitlets\config\application.py", line 846, in launch_instance
  File "main.py", line 184, in start
  File "main.py", line 157, in make_flashcard
  File "main.py", line 132, in take_screenshot
  File "main.py", line 60, in play_camera_sound
  File "playsound.py", line 35, in _playsoundWin
  File "playsound.py", line 31, in winCommand
playsound.PlaysoundException: 
    Error 275 for command:
        open "sounds\shutter-40453.mp3" alias playsound_0.3972068701089335
    Cannot find the specified file.  Make sure the path and filename are correct.
```

however [the relevant code to play a sound](https://github.com/metaperl/flash-goban/blob/main/main.py#L50) works just fine when running `main.py` from a windows cmd.exe prompt.

**Comment by bwoodsend** [comment]
> however [the relevant code to play a sound](https://github.com/metaperl/flash-goban/blob/main/main.py#L50) works just fine when running main.py from a windows cmd.exe prompt.

It doesn't really. `cd` to your home directory (or anywhere out of the way will do) then run `python C:\path\to\main.py`.

**Comment by metaperl** [comment]
> It doesn't really

you're right. i should've said: [the relevant code to play a sound](https://github.com/metaperl/flash-goban/blob/main/main.py#L50) works just fine when running main.py from a windows cmd.exe prompt that is `chdir`ed to the root of the source code repo.

This brings up some questions about current working directory in the generated `.exe` versus me running from the command prompt. is this documented in the manual?

**Reply by bwoodsend** [reply]
Hmm, you're missing the point slightly. Your code needs to become runnable from any working directory otherwise it'll fall to pieces even without PyInstaller's help. [Reference files relative to `__file__`](https://pyinstaller.org/en/v5.4.1/runtime-information.html#using-file) so that `python C:\path\to\main.py` works anywhere then figure out which `--add-data` command you need to give to PyInstaller in order to include that `sounds` directory.

**Reply by metaperl** [reply]
i see. well it works fine now that i removed `--onefile` from [the spec file](https://github.com/metaperl/flash-goban/blob/main/make-exe.spec). but will take note of the comments about `__file__` should things go awry.

**Comment by metaperl** [comment]
i've actually figured this out. I created the spec file using `--onefile` and so no folders were packaged. now the generated `.exe` in the dist directory runs fine.

But now i have the issue of how to provide an executable that contains all the folders and that unpacks them when the `.exe` is invoked.
